app/controllers/application_controller.rb

```diff
@@ -1,27 +1,47 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::API
-  before_action :authenticate!
+  # The HMPPS Auth role required for 'write' endpoints
+  WRITE_ROLE = "ROLE_COMPLEXITY_OF_NEED"
+
+  before_action :authorise_read!
 
 private
 
-  def authenticate!
-    access_token = parse_access_token(request.headers["AUTHORIZATION"])
+  def authorise_read!
+    if token.nil?
+      render_bad_token
+    elsif token.has_scope?("read") == false
+      render_forbidden "You need the scope 'read' to use this endpoint"
+    end
+  end
 
-    token = HmppsApi::Oauth::Token.new(access_token)
-    unless token.valid_token_with_scope?("read")
-      render_unauthorized("Valid authorisation token required")
+  def authorise_write!
+    if token.nil?
+      render_bad_token
+    elsif (token.has_scope?("write") && token.has_role?(WRITE_ROLE)) == false
+      render_forbidden "You need the role '#{WRITE_ROLE}' with scope 'write' to use this endpoint"
     end
   end
 
-  def parse_access_token(auth_header)
-    return nil if auth_header.nil?
-    return nil unless auth_header.starts_with?("Bearer")
+  def token
+    @token ||= begin
+                 auth_header = request.headers["Authorization"]
+
+                 if auth_header.present? && auth_header.starts_with?("Bearer")
+                   access_token = auth_header.split.last
+                   HmppsApi::Oauth::Token.new(access_token)
+                 end
+               rescue JWT::DecodeError, NoMethodError
```

Comment:
why are we rescuing NoMethodError here - looks scary

Reply:
Actually there's a recognised pattern for this - use RescueFrom in the controller, and don't catch it here...